infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/util/GatewayConfigmapLabels.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes.util;
+
+import static java.util.stream.Collectors.toMap;
+import static org.eclipse.che.workspace.infrastructure.kubernetes.namespace.KubernetesObjectUtil.isLabeled;
+
+import com.google.common.collect.ImmutableMap;
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Map.Entry;
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+/** Little utility bean helping with Gateway ConfigMaps labels. */
+@Singleton
+public class GatewayConfigmapLabels {
+
+  private final Map<String, String> labels;
+
+  @Inject
+  public GatewayConfigmapLabels(
+      @Named("che.infra.kubernetes.single_host.gateway.configmap.labels") String[] labelsProperty) {
```

Comment:
What is this sorcery? :) Is this a standard feature of Guice that it splits a string by a comma to satisfy the String[] arg? IMHO, it would be more self-documenting if we did the parsing in the body of this method...

Reply:
yes, I got inspiration from this https://github.com/eclipse/che/blob/master/infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/secret/SecretAsContainerResourceProvisioner.java#L57 but we're using pure String with Splitter as well https://github.com/eclipse/che/blob/master/infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/server/external/IngressServerExposer.java#L59

Reply:
I don't have strong opinion here. @skabashnyuk @mshaposhnik ?

Reply:
I vote for `parsing in the body`. It brings more space for maneuvering.

Reply:
done